x-pack/legacy/plugins/apm/public/components/shared/ServiceAlertTrigger/index.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useEffect } from 'react';
+import {
+  EuiSpacer,
+  EuiFlexGrid,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiText
+} from '@elastic/eui';
+import { useUrlParams } from '../../../hooks/useUrlParams';
+
+interface FieldProps {
+  title: React.ReactNode;
+  field: React.ReactNode;
+  name: string;
+}
+
+interface Props {
+  alertTypeName: string;
+  setAlertParams: (key: string, value: any) => void;
+  setAlertProperty: (key: string, value: any) => void;
+  fields: FieldProps[];
+  defaults: Record<string, any>;
+}
+
+const Field = (props: FieldProps) => {
+  const { title, field } = props;
+
+  return (
+    <EuiFlexItem>
+      <EuiFlexGroup gutterSize="xs" direction="column">
+        <EuiFlexItem grow={false}>
+          <EuiText size="xs">
+            <h4>{title}</h4>
+          </EuiText>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>{field}</EuiFlexItem>
+      </EuiFlexGroup>
+    </EuiFlexItem>
+  );
+};
+
+export function ServiceAlertTrigger(props: Props) {
+  const { urlParams } = useUrlParams();
+
+  const {
+    fields,
+    setAlertParams,
+    setAlertProperty,
+    alertTypeName,
+    defaults
+  } = props;
+
+  const params: Record<string, any> = {
+    ...defaults,
+    serviceName: urlParams.serviceName!,
+    transactionType: urlParams.transactionType!
+  };
+
+  useEffect(() => {
+    setAlertProperty(
+      'name',
+      `${alertTypeName} | ${params.serviceName} | ${params.transactionType}`
+    );
+    setAlertProperty('tags', [
+      'apm',
+      `service.name:${params.serviceName}`,
+      `transaction.type:${params.transactionType}`
```

Comment:
I think we should use the ecs constants:

```suggestion
      `${SERVICE_NAME}:${params.serviceName}`,
      `${TRANSACTION_TYPE}:${params.transactionType}`
```

Ntw. what do you think about renaming elasticsearch_fieldnames.ts to ecs.ts ?

Reply:
It's not all ECS (unfortunately). ie, `service.environment` is not in the ECS specification. Same goes for `error.grouping_key` and a couple of others. So maybe `elasticsearch_fieldnames` is good enough for now?

Reply:
btw, I'm removing the `service.name` and `transaction.type` tags. We're not doing anything wiht lookups now, and in the future (IIRC) we can use the params.